include/envoy/buffer/buffer.h

```diff
@@ -179,6 +182,118 @@ class Instance {
    *   for failure. If the call is successful, errno_ shouldn't be used.
    */
   virtual Api::SysCallResult write(int fd) PURE;
+
+  /**
+   * Copy an integral out of the buffer.
+   * @param start supplies the buffer index to start copying from.
+   * @param Size how many bytes to read out of the buffer.
+   * @param Endianness specifies the byte order to use when decoding the integral.
+   */
+  template <typename T, ByteOrder Endianness = ByteOrder::Host, size_t Size = sizeof(T)>
```

Comment:
Nice. I really like this set of changes. Can we please move this change (and the associated Thrift changes) into a separate change? We should make sure the new buffer functions have explicit tests in buffer_impl_test. Let's get that change first and then we can merge master into this change and get this in. Thank you!